### PR TITLE
Linting

### LINKED ISSUE #264: Make `deno lint` Pass
## Summary

What: If you run `deno lint`, we will most likely have a whole bunch of errors. This issue to fix these errors by correctly formatting code, and abiding by Deno's standards 

Why: It would be good to make sure the code follows either: Deno's best practices, coding standards, or general best practices. It's a Deno project so it makes sense to utilise their tools on making the code Deno-like

## Acceptance Criteria

- `deno lint` is added to the `master.yml` CI, as a new block (similar to how we use `deno fmt --check` on some repositories)
- `deno lint` produces no errors

## Resources

[Deno Lint Documentation](https://github.com/denoland/deno_lint)

### LINKED ISSUE #275: Type for request
## Summary

What: typing for the request property that is present on Drash.Http.Resource

Why: It would be nice for users to have a type to use in helper functions that they may want to write.  It would also help code completion when building a Drash.Http.Resource

### PR DESCRIPTION
**Pull request by crookse**
Closes #264 
Closes #275

Notes:

* this pr ignores `mod.ts` and `members.ts`. i dont think these files should be linted.
    * `mod.ts` is our own custom way of implementing our `Drash` namespace and throws a bunch of namespace linting errors. maybe in v2 we'll move towards a module-based system.
    * `members.ts` is just for testing and is only used internally for our needs. nobody outside of our team should be using it.
* this pr ignores some `any` lines. we should come back to these at a later time imo.
* this pr unifies all of the file markers (the `// FILE MARKER ////` lines)

### CODE REVIEW

File: src/interfaces/parsed_request_body.ts [hunk 1]
```diff
@@ -13,6 +13,7 @@
  *         The data passed in the body of the request.
  */
 export interface ParsedRequestBody {
-  content_type: any | undefined;
+  content_type: string;
+  // deno-lint-ignore no-explicit-any
   data: any | undefined;
```
Comment (Guergeiro) [inline src/interfaces/parsed_request_body.ts:18]: Shouldn't we use `unknown` here?

File: src/services/http_service.ts [hunk 2]
```diff
@@ -83,6 +87,7 @@ export class HttpService {
       let fileParts = filePath.split(".");
       filePath = fileParts.pop();
 
+      // deno-lint-ignore no-explicit-any
       const database: any = Drash.Dictionaries.MimeDb;
```
Comment (Guergeiro) [inline src/services/http_service.ts:91]: Again, isn't `unknown` a good fit for this?

Comment (crookse) [reply src/services/http_service.ts]: i defined an interface for the MIME DB dictionary: 3ac322e

File: src/compilers/template_engine.ts [hunk 3]
```diff
@@ -34,17 +38,17 @@ export class TemplateEngine {
    * @param any data
    *     The data that should be rendered with the template.
    */
-  public render(template: string, data: any): string {
-    let code: any = "with(obj) { var r=[];\n";
-    let cursor: any = 0;
+  public render(template: string, data: unknown): string {
```
Comment (ebebbington) [inline src/compilers/template_engine.ts:41]: Wonder if it's worth we give an example usage line as we can't properly type the `data` param?

File: src/core_loggers/console_logger.ts [hunk 4]
```diff
@@ -33,7 +41,10 @@ export class ConsoleLogger extends Logger {
    * @return string
    *     Returns the log message which is used for unit testing purposes.
```
Comment (ebebbington) [inline src/core_loggers/console_logger.ts:42]: To also include something like `and returns void if blah blah blah`?

File: src/core_loggers/file_logger.ts [hunk 5]
```diff
@@ -45,7 +45,10 @@ export class FileLogger extends Logger {
    * @return string|void
    *     Returns the log message which is used for unit testing purposes.
    */
-  public write(logMethodLevelDefinition: any, message: string): string | void {
+  public write(
+    logMethodLevelDefinition: Drash.Interfaces.LogLevelStructure,
+    message: string,
+  ): string | void {
```
Comment (ebebbington) [inline src/core_loggers/file_logger.ts:51]: To also include something like `and returns void if blah blah blah`?

File: src/http/response.ts [hunk 6]
```diff
@@ -114,7 +150,10 @@ export class Response {
    * @return string|boolean
    *     The html content of the view, or false if the `views_path` is not set.
    */
-  public render(...args: any): string | boolean {
+  public render(
+    // deno-lint-ignore no-explicit-any
+    ...args: any
```
Comment (ebebbington) [inline src/http/response.ts:155]: `...args: unknown` just to match the doc block?

File: src/http/response.ts [hunk 7]
```diff
@@ -322,14 +367,19 @@ export class Response {
   public redirect(
     httpStatusCode: number,
     location: string,
-  ): { status: number; headers: Headers; body: any } {
+  ): Drash.Interfaces.ResponseOutput {
```
Comment (ebebbington) [inline src/http/response.ts:400]: Has this doc block been updated to. use. `Drash.Interfaces.ResponseOutput`?

File: src/http/response.ts [hunk 8]
```diff
@@ -15,7 +46,7 @@ export class Response {
    * @description
    *     A property to hold the body of this response.
    *
-   * @property any body
+   * @property string body
    */
   public body: any = "";
```
[the remark this answers is not in the document]
Comment (crookse) [reply src/http/response.ts]: i addressed this here: d06207c

File: src/http/server.ts [hunk 9]
```diff
@@ -84,25 +94,29 @@ export class Server {
...
+      | ((request: Drash.Http.Request) => void)
     >;
     after_request?: Array<
-      | ((request: any, response: Drash.Http.Response) => Promise<void>)
-      | ((request: any, response: Drash.Http.Response) => void)
+      | ((
+        request: Drash.Http.Request,
+        response: Drash.Http.Response,
+      ) => Promise<void>)
+      | ((request: Drash.Http.Request, response: Drash.Http.Response) => void)
```
Comment (ebebbington) [inline src/http/server.ts:109]: Maybe we could create an interface and then we could type it in the docs?

File: src/http/server.ts [hunk 10]
```diff
@@ -315,17 +349,19 @@ export class Server {
...
-    request: any,
-    error: any,
-    resource: Drash.Http.Resource | any = {},
-    response: Drash.Http.Response | any = {},
-  ): Promise<any> {
+    request: Drash.Http.Request,
+    error: Drash.Exceptions.HttpException,
+    resource: Drash.Http.Resource | null = null,
+    response: Drash.Http.Response | null = null,
+  ): Promise<Drash.Interfaces.ResponseOutput> {
```
Comment (ebebbington) [inline src/http/server.ts:356]: Has the doc block. been updated to reflect this change?

File: src/http/server.ts [hunk 11]
```diff
@@ -385,7 +425,9 @@ export class Server {
    *     Returns the response as stringified JSON. This is only used for unit
    *     testing purposes.
    */
-  public handleHttpRequestForFavicon(request: any): string {
+  public handleHttpRequestForFavicon(
+    request: Drash.Http.Request,
+  ): Drash.Interfaces.ResponseOutput {
```
Comment (ebebbington) [inline src/http/server.ts:430]: Has the doc block been updated to reflect this?

File: src/http/server.ts [hunk 12]
```diff
@@ -426,65 +475,41 @@ export class Server {
    *     Returns the response as stringified JSON. This is only used for unit
    *     testing purposes.
    */
-  public handleHttpRequestForStaticPathAsset(request: any): any {
+  public async handleHttpRequestForStaticPathAsset(
+    request: Drash.Http.Request,
+  ): Promise<Drash.Interfaces.ResponseOutput> {
```
Comment (ebebbington) [inline src/http/server.ts:480]: Has the doc block been updated to reflect this?

File: src/http/server.ts [hunk 13]
```diff
@@ -684,7 +716,7 @@ export class Server {
    * @return void
    */
   protected async executeMiddlewareServerLevelBeforeRequest(
-    request: any,
+    request: Drash.Http.Request,
```
Comment (ebebbington) [inline src/http/server.ts:719]: Has the doc block been updated to reflect this?

File: src/http/server.ts [hunk 14]
```diff
@@ -706,7 +738,7 @@ export class Server {
    * @return void
    */
   protected async executeMiddlewareServerLevelAfterRequest(
-    request: any,
+    request: Drash.Http.Request,
```
Comment (ebebbington) [inline src/http/server.ts:741]: Has the doc block been updated to reflect this?

File: src/http/server.ts [hunk 15]
```diff
@@ -744,11 +776,13 @@ export class Server {
    *
    *     Returns `undefined` if a `Drash.Http.Resource` object can't be matched.
    */
-  protected getResourceClass(request: any): Drash.Http.Resource | undefined {
+  protected getResourceClass(
+    request: Drash.Http.Request,
+  ): Drash.Interfaces.Resource | undefined {
```
Comment (ebebbington) [inline src/http/server.ts:781]: Has the doc block been updated to reflect this?

File: src/http/server.ts [hunk 16]
```diff
@@ -782,28 +816,32 @@ export class Server {
    * @return boolean
    *     Returns true if the request targets a static path.
    */
-  protected requestTargetsStaticPath(request: any): boolean {
+  protected requestTargetsStaticPath(serverRequest: ServerRequest): boolean {
```
Comment (ebebbington) [inline src/http/server.ts:819]: Has the doc block been updated to reflect this?

File: src/services/http_service.ts [hunk 17]
```diff
@@ -32,7 +36,10 @@ export class HttpService {
    *     False if the request doesn't accept any of the passed in types,
    *     or the content type that was matches
    */
-  public accepts(request: any, type: string | string[]): boolean | string {
+  public accepts(
```
Comment (ebebbington) [inline src/services/http_service.ts:39]: Has the doc block been updated to reflect this?

File: src/http/response.ts (hunk 6 above)
Comment (crookse) [reply src/http/response.ts]: addressed: ba8108d

File: src/http/response.ts (hunk 7 above)
Comment (crookse) [reply src/http/response.ts:404]: 9d19552

File: src/http/server.ts (hunk 16 above)
Comment (crookse) [reply src/http/server.ts:814]: 3565ceb

File: src/http/server.ts (hunk 15 above)
Comment (crookse) [reply src/http/server.ts:776]: 3565ceb

File: src/http/server.ts (hunk 14 above)
Comment (crookse) [reply src/http/server.ts:736]: 3565ceb

File: src/http/server.ts (hunk 13 above)
Comment (crookse) [reply src/http/server.ts:714]: 3565ceb

File: src/services/http_service.ts (hunk 17 above)
Comment (crookse) [reply src/services/http_service.ts:44]: c6c3084 and b73933a

File: src/core_loggers/console_logger.ts (hunk 4 above)
Comment (crookse) [reply src/core_loggers/console_logger.ts:42]: 47c55d3

File: src/compilers/template_engine.ts (hunk 3 above)
Comment (crookse) [reply src/compilers/template_engine.ts:48]: 1a3e7f2

File: src/core_loggers/file_logger.ts (hunk 5 above)
Comment (crookse) [reply src/core_loggers/file_logger.ts]: 47c55d3

File: src/http/server.ts (hunk 9 above)
Comment (crookse) [reply src/http/server.ts:109]: i think we should do this in another issue. it was hard as fuck to type

File: src/http/server.ts (hunk 10 above)
Comment (crookse) [reply src/http/server.ts:351]: updated

File: src/http/server.ts (hunk 11 above)
Comment (crookse) [reply src/http/server.ts:425]: updated

File: src/http/server.ts (hunk 12 above)
Comment (crookse) [reply src/http/server.ts:475]: updated

File: src/interfaces/parsed_request_body.ts (hunk 1 above)
Comment (crookse) [reply src/interfaces/parsed_request_body.ts:18]: i tried and failed to make the code compile

**Pull request merged**